src/acquisition/fluview/fluview_update.py

```diff
@@ -130,398 +130,422 @@
+    `num_age_2` = coalesce(%s, `num_age_2`),
+    `num_age_3` = coalesce(%s, `num_age_3`),
+    `num_age_4` = coalesce(%s, `num_age_4`),
+    `num_age_5` = coalesce(%s, `num_age_5`)
+    """
+
+    # insert each row
+    insert = cnx.cursor()
+    for row in entries:
+        lag = delta_epiweeks(row["epiweek"], issue)
+        args = [
+            row["n_ili"],
+            row["n_patients"],
+            row["n_providers"],
+            row["wili"],
+            row["ili"],
+            row["age0"],
+            row["age1"],
+            row["age2"],
+            row["age3"],
+            row["age4"],
+            row["age5"],
+        ]
+        ins_args = [date, issue, row["epiweek"], row["location"], lag] + args
+        upd_args = [date] + args
+        insert.execute(sql, ins_args + upd_args)
+
+    # cleanup
+    insert.close()
+    if test_mode:
+        print("test mode, not committing")
+        rows2 = rows1
+    else:
+        cnx.commit()
+        rows2 = get_rows(cnx)
+    print("rows after: %d (added %d)" % (rows2, rows2 - rows1))
+    cnx.close()
+
 
 def main():
-  # args and usage
-  parser = argparse.ArgumentParser()
-  parser.add_argument(
-    '--test',
-    action='store_true',
-    help='do dry run only, do not update the database'
-  )
-  parser.add_argument(
-    '--file',
-    type=str,
-    help='load an existing zip file (otherwise fetch current data)'
-  )
-  parser.add_argument(
-    '--issue',
-    type=int,
-    help='issue of the file (e.g. 201740); used iff --file is given'
-  )
-  args = parser.parse_args()
-
-  if (args.file is None) != (args.issue is None):
-    raise Exception('--file and --issue must both be present or absent')
-
-  date = datetime.datetime.now().strftime('%Y-%m-%d')
-  print('assuming release date is today, %s' % date)
-
-  if args.file:
-    update_from_file(args.issue, date, args.file, test_mode=args.test)
-    update_from_file_clinical(args.issue, date, args.file, test_mode=args.test)
-    # TODO: header row has changed for public health lab data
-    # update_from_file_public(args.issue, date, args.file, test_mode=args.test)
-  else:
-    issue, files = fluview.save_latest(path='flu_data')
-    for filename in files:
-      update_from_file(issue, date, filename, test_mode=args.test)
-      update_from_file_clinical(issue, date, filename, test_mode=args.test)
-      # TODO: header row has changed for public health lab data
-      # update_from_file_public(issue, date, filename, test_mode=args.test)
-
-if __name__ == '__main__':
-  main()
+    # args and usage
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--test", action="store_true", help="do dry run only, do not update the database"
+    )
+    parser.add_argument(
+        "--file", type=str, help="load an existing zip file (otherwise fetch current data)"
+    )
+    parser.add_argument(
+        "--issue", type=int, help="issue of the file (e.g. 201740); used iff --file is given"
```

Comment:
hmmmm I'm skeptical about these. if it were just one, fine, but with three of them it's helpful to have whitespace highlight argument alignment